src/main/java/com/zenith/Proxy.java

```diff
@@ -156,6 +156,35 @@ public void start() {
             // ensure we are continuously updating the tablist even on servers that don't frequently send updates
             SCHEDULED_EXECUTOR_SERVICE.scheduleAtFixedRate(this::tablistUpdate, 20L, 3L, TimeUnit.SECONDS);
             SCHEDULED_EXECUTOR_SERVICE.submit(this::updatePrioBanStatus);
+            // time warning
+            SCHEDULED_EXECUTOR_SERVICE.scheduleAtFixedRate(() -> {
+                try {
+                    if (isOnlineOn2b2tForAtLeastDuration(Duration.ofSeconds(60))) {
+                        long onlineSeconds = Instant.now().getEpochSecond() - connectTime.getEpochSecond();
+                        if (onlineSeconds > (21600 - (600))) { // 28800 - 600 (6h - 10 minutes)
+                            for (int i=0; i<10; i++) { //Sends the message 10 times, so you don't miss it. It also sends it every minute from then
+                                Proxy.getInstance().getActiveConnections().forEach(connection -> {
```

Comment:
just sending this to the active player is fine: 

```
final ServerConnection playerConnection = this.currentPlayer.get();
if (playerConnection != null) // we have a player logged in, send any warning packets to this connection
  playerConnection.send(...)
```

And it'd be better to just check if there is an active player or not near the top of this runnable to skip most of the logic if not.

`if (!this.isConnected() || !this.hasActivePlayer()) return; // no active player - no need to do any in-game warnings`